server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -248,11 +249,12 @@ public InternalEngine(EngineConfig engineConfig) {
         Translog translog = null;
         ExternalReaderManager externalReaderManager = null;
         ElasticsearchReaderManager internalReaderManager = null;
-        EngineMergeScheduler scheduler = null;
+        MergeScheduler scheduler = null;
         boolean success = false;
         try {
             this.lastDeleteVersionPruneTimeMSec = engineConfig.getThreadPool().relativeTimeInMillis();
-            mergeScheduler = scheduler = new EngineMergeScheduler(engineConfig.getShardId(), engineConfig.getIndexSettings());
+            mergeScheduler = createMergeScheduler(engineConfig.getShardId(), engineConfig.getIndexSettings());
+            scheduler = (MergeScheduler) mergeScheduler;
```

Comment:
nit: I'd prefer to add a `getMergeScheduler` method to `ElasticsearchMergeScheduler`, which could then return this (or a separate object if desired). Happy to not do this though, it is a very minor nit.